cilium-dbg/cmd/encrypt_status.go

```diff
@@ -47,81 +55,132 @@ func init() {
 	command.AddOutputOption(encryptStatusCmd)
 }
 
-func getXfrmStats(mountPoint string) (int, map[string]int) {
+func getEncryptionMode() (models.EncryptionStatus, error) {
+	params := daemon.NewGetHealthzParamsWithTimeout(timeout)
+	params.SetBrief(&brief)
+	resp, err := client.Daemon.GetHealthz(params)
+	if err != nil {
+		return models.EncryptionStatus{}, err
+	}
+
+	enc := resp.Payload.Encryption
+	status := models.EncryptionStatus{
+		Mode: enc.Mode,
+	}
+	switch enc.Mode {
+	case models.EncryptionStatusModeIPsec:
+		if err := dumpIPsecStatus(&status); err != nil {
+			return models.EncryptionStatus{}, err
+		}
+	case models.EncryptionStatusModeWireguard:
+		dumpWireGuardStatus(enc, &status)
+	}
+	return status, nil
+}
+
+func dumpIPsecStatus(status *models.EncryptionStatus) error {
```

Comment:
Is there a good reason to pass a pointer to the function and modify that pointer instead having a function like
```suggestion
func getIPsecStatus() (models.EncryptionStatus, error) {
```

that builds the status from scratch without having to deal with pointers here? That would feel more idiomatic to me.

Reply:
My initial intention was not to do refactoring but I fully agree with you.
Done, thanks!